File: content/en/04-llms/llms/_solutions/06_word-games/06_word-games-app.py
import random

import chatlas
from pyhere import here
from shiny import App, ui
from shinychat import Chat, chat_ui

# Picks a random word for the model to keep secret from a list.
word = random.choice(here("data/words.txt").read_text().splitlines())

system_prompt = f"""
We are playing a word guessing game. The secret word is "{word}".

Never say or otherwise reveal the secret word before the user guesses it.
Give the user an initial clue and then only answer their questions with yes or no.
When they win, use lots of emojis.
"""

app_ui = ui.page_fillable(
    chat_ui("chat", placeholder="""Say "Let's play" to get started!""")
)


def server(input, output, session):
    client = chatlas.ChatPosit(system_prompt=system_prompt)
    chat = Chat("chat", client=client)


app = App(app_ui, server)

File: content/en/04-llms/llms/_solutions/06_word-games/06_word-games-app.R
library(shiny)
library(ellmer)
library(shinychat)

# Picks a random word for the model to keep secret from a list.
word <- sample(readLines(here::here("data/words.txt")), 1)

system_prompt <- interpolate(
  r"--(
We are playing a word guessing game. The secret word is "{{ word }}".

Never say or otherwise reveal the secret word before the user guesses it.
Give the user an initial clue and then only answer their questions with yes or no.
When they win, use lots of emojis.
)--"
)


ui <- page_chat(
  "Word Games",
  placeholder = r"(Say "Let's play" to get started!)"
)

server <- function(input, output, session) {
  client <- chat_posit(system_prompt = system_prompt)
  chat_server("chat", client)
}

shinyApp(ui, server)
